contracts/Strategy.sol

```diff
@@ -105,6 +158,119 @@ contract Strategy is BaseStrategy {
+        _invest();
+    }
+
+    function _claimCometRewards() internal {
+        rewardsContract.claim(address(cToken), address(this), true);
+    }
+
+    function _sellRewards() internal {
+        //check for Trade Factory implementation or that Uni fees are not set
+        if(tradeFactory != address(0) || compToEthFee == 0) return;
+
+        uint256 _comp = IERC20(comp).balanceOf(address(this));
+
+        if (_comp > minCompToSell) {
+
+            if(address(asset) == weth) {
+                ISwapRouter.ExactInputSingleParams memory params =
+                    ISwapRouter.ExactInputSingleParams(
+                        comp, // tokenIn
+                        address(asset), // tokenOut
+                        compToEthFee, // comp-eth fee
+                        address(this), // recipient
+                        block.timestamp, // deadline
+                        _comp, // amountIn
+                        0, // amountOut
+                        0 // sqrtPriceLimitX96
+                    );
+
+                router.exactInputSingle(params);
+            
+            } else {
+                bytes memory path =
+                    abi.encodePacked(
+                        comp, // comp-ETH
+                        compToEthFee,
+                        weth, // ETH-asset
+                        ethToWantFee,
+                        address(asset)
+                    );
+
+                // Proceeds from Comp are not subject to minExpectedSwapPercentage
+                // so they could get sandwiched if we end up in an uncle block
+                router.exactInput(
+                    ISwapRouter.ExactInputParams(
+                        path,
+                        address(this),
+                        block.timestamp,
+                        _comp,
+                        0
+                    )
+                );
+            }
+        }
+    }
+
+    /*
+    * Get the current reward for supplying APR in Compound III
+    * @param rewardTokenPriceFeed The address of the reward token (e.g. COMP) price feed
+    * @param newAmount Any amount that will be added to the total supply in a deposit
+    * @return The reward APR in USD as a decimal scaled up by 1e18
+    */
+    function getRewardAprForSupplyBase(address rewardTokenPriceFeed, int newAmount) public view returns (uint) {
+        uint rewardToSuppliersPerDay = cToken.baseTrackingSupplySpeed() * SECONDS_PER_DAY * BASE_INDEX_SCALE / BASE_MANTISSA;
+        if(rewardToSuppliersPerDay == 0) return 0;
+
+        uint rewardTokenPriceInUsd = getCompoundPrice(rewardTokenPriceFeed);
+        uint assetPriceInUsd = getCompoundPrice(cToken.baseTokenPriceFeed());
+        uint assetTotalSupply = uint256(int256(cToken.totalSupply()) + newAmount);
+        return rewardTokenPriceInUsd * rewardToSuppliersPerDay / (assetTotalSupply * assetPriceInUsd) * DAYS_PER_YEAR;
+    }
+
+    function getPriceFeedAddress(address asset) public view returns (address) {
```

Comment:
This is an unlikely scenario and doesnt apply for usdc, but its possible in the future that the reward token (comp) would be the same as the baseToken and that would cause this to fail, because baseToken doesnt have a stuct to call .priceFeed.

If it matters can add a check in the first line that if(asset == basToken) return cToken.baseTokenPriceFeed()